source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -241,59 +241,61 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
         columnDefinition.Width(CalculateColumnWidth(true, false, false, isWidthUnset, pixelWidth, width));
     }
 
+    rtxaml::Controls::Image CreateBackgroundImage(rtrender::AdaptiveRenderContext const& renderContext, winrt::hstring const& url)
+    {
+        // TODO: if url is invalid, it will bitmapImage.UriSource(nullptr) will throw, right?
+        rtxaml::Controls::Image backgroundImage;
+
+        // GetUrlFromString will throw if url is not in correct format
+        auto imageUrl = GetUrlFromString(renderContext.HostConfig(), url);
+
+        rtxaml::Media::Imaging::BitmapImage bitmapImage{};
+        bitmapImage.UriSource(imageUrl);
+
+        backgroundImage.Source(bitmapImage);
+
+        return backgroundImage;
+    }
+
     void ApplyBackgroundToRoot(rtxaml::Controls::Panel const& rootPanel,
-                               rtom::AdaptiveBackgroundImage const& backgroundImage,
-                               rtrender::AdaptiveRenderContext const& renderContext,
-                               rtrender::AdaptiveRenderArgs const& renderArgs)
+                               rtom::AdaptiveBackgroundImage const& adaptiveBackgroundImage,
+                               rtrender::AdaptiveRenderContext const& renderContext)
     {
         // In order to reuse the image creation code paths, we simply create an adaptive card
         // image element and then build that into xaml and apply to the root.
         // TODO: it might be an overkill to render apdativeImage for background image
         // TODO: we should probably create a routine to do that without adaptive image. Should be simple enough.
-        rtom::AdaptiveImage adaptiveImage;
-        adaptiveImage.Url(backgroundImage.Url());
-
-        if (const auto elementRenderer = renderContext.ElementRenderers().Get(L"Image"))
+        if (const auto backgroundImage = CreateBackgroundImage(renderContext, adaptiveBackgroundImage.Url()))
```

Comment:
I'm not totally sure how to handle errors here. If url is invalid - meaning wrong format - we will throw in **GetUrlFromString** above.
@paulcam206 what do you think?

Reply:
we should probably catch the exception out of GetUrlFromString, log a warning, and not attempt to render the background...